_posts/2021-10-01-gs_00004_basic_shell_script.md

```diff
@@ -0,0 +1,318 @@
+Whenever you have to do something regularly you should ask yourself if it is
+possible to automate it. This is more or less what a **Script** does. It calls a
+sequence of commands for you. We want to give you two examples that might be
+useful in the real world. Before that, we have to clarify the term **Shell**.
+
+</div>
+
+# Shell, Terminal, Console ...
+
+All these terms are very confusing for beginners as well as advanced engineers.
+Regularly, they are used interchangeably. Let's try to make them as clear as
+possible. In the very beginning, computers were huge and usually had no user
+interface. You needed some extra **physical device** that was connected to the
+computer, which was called **Console**. In contrast, the **Terminal** is not
+something physical. It is simply a text input and output environment. So far so
+good. You have a physical device (Console) with a keyboard or something similar
+and kind of a program where you can type some text in (Terminal). But there is
+one step more. Something needs to read the text you are typing and **interpret**
+its meaning. The software that is doing this task is called **Shell**. If you
+enter a command like "ls", the shell _understands_ that you want to get the
+content of the current folder. It executes the command and returns the output to
+the terminal.
+
+As you can have different physical devices, you can also have different
+terminals and shells. A different terminal might look different from another
+one. A different shell might interpret different commands. Wait ... What? Yeah,
+you heard correctly! Commands that work in one shell might work differently or
+not at all in another. But don't worry! Just have it in mind for now! By the
+way, the shell we used in the previous posts is called **Bash**. We might talk
+about other shells in future posts, but for now, we stay with this commonly used
+shell.
+
+# Your first Bash Script
+
+Every morning the same procedure. You go to work, get your first coffee and then
+open all required software for your job: Email client, meeting software,
+browser, music app, and so on. Probably it might be different depending on your
+daily work schedule. Wouldn't it be cool that at least the software opening part
+is done automatically? Let's take it as an example for the first script and open
+a browser. You already know how to do it with a terminal command, don't you? To
+write your first script, open a new file:
+
+    $ nano open_firefox.sh
+
+<div class='note-normal' markdown="1">
+_Note: '.sh' is the typical file extension for shell scripts_.
+</div>
+
+Enter the following text:
+
+    #!/bin/bash
+
+    # Print something to the terminal
+    echo "Greetings from Bits&Volts (bitsandvolts.org)"
+
+    # Open firefox
+    firefox &
+
+Press CTRL-X to close and 'y' (yes) to save the file. Before we run it, let's
+talk about it. The first line starting with the '#!' tells by what the script
```

Comment:
Did you mean this:
```suggestion
talk about it. The first line starting with the '#!' tells by what shell the script
```

Reply:
does not need to be shell. recheck

Reply:
Yes, sheabang could be anything. Maybe you can add a footnote or something, that in our case we use the sheabang to select a shell interpreter.

But decide for yourself; you can also ignore my suggestion.

Reply:
let's discuss